drop-storage/src/lib.rs

```diff
@@ -1,630 +1,532 @@
 
-        sqlx::migrate!("./migrations")
-            .run(&mut conn.acquire().await?)
-            .await
-            .map_err(|e| error::Error::InternalError(format!("Failed to run migrations: {e}")))?;
+        let mut conn = pool.get()?;
+        MIGRATIONS
+            .to_latest(&mut conn)
+            .map_err(|e| Error::InternalError(format!("Failed to run migrations: {e}")))?;
 
         Ok(Self {
             _logger: logger,
-            conn,
+            pool,
         })
     }
 
-    pub async fn insert_transfer(&self, transfer: &TransferInfo) -> Result<()> {
+    pub fn insert_transfer(&self, transfer: &TransferInfo) -> Result<()> {
         let transfer_type_int = match &transfer.files {
             TransferFiles::Incoming(_) => TransferType::Incoming as u32,
             TransferFiles::Outgoing(_) => TransferType::Outgoing as u32,
         };
 
-        let tid = transfer.id.hyphenated();
+        let tid = transfer.id.hyphenated().to_string();
 
-        let mut conn = self.conn.begin().await?;
+        let mut conn = self.pool.get()?;
+        let conn = conn.transaction()?;
 
-        sqlx::query!(
-            "INSERT OR IGNORE INTO transfers (id, peer, is_outgoing) VALUES (?1, ?2, ?3)",
-            tid,
-            transfer.peer,
-            transfer_type_int,
-        )
-        .execute(&mut *conn)
-        .await?;
+        conn.execute(
+            "INSERT INTO transfers (id, peer, is_outgoing) VALUES (?1, ?2, ?3)",
+            params![tid, transfer.peer, transfer_type_int],
+        )?;
 
         match &transfer.files {
             TransferFiles::Incoming(files) => {
                 for file in files {
-                    Self::insert_incoming_path(&mut conn, transfer.id, file).await?;
+                    Self::insert_incoming_path(&conn, transfer.id, file)?;
                 }
             }
             TransferFiles::Outgoing(files) => {
                 for file in files {
-                    Self::insert_outgoing_path(&mut conn, transfer.id, file).await?;
+                    Self::insert_outgoing_path(&conn, transfer.id, file)?;
                 }
             }
         }
 
-        conn.commit().await.map_err(error::Error::DBError)
+        conn.commit()?;
+
+        Ok(())
     }
 
-    async fn insert_incoming_path(
-        conn: &mut SqliteConnection,
+    fn insert_incoming_path(
+        conn: &Transaction<'_>,
         transfer_id: Uuid,
         path: &TransferIncomingPath,
     ) -> Result<()> {
-        let tid = transfer_id.hyphenated();
-
-        sqlx::query!(
-            r#"
-            INSERT INTO incoming_paths (transfer_id, relative_path, path_hash, bytes)
-            VALUES (?1, ?2, ?3, ?4)
-            ON CONFLICT DO NOTHING
-            "#,
-            tid,
-            path.relative_path,
-            path.file_id,
-            path.size,
-        )
-        .execute(conn)
-        .await?;
+        let tid = transfer_id.hyphenated().to_string();
```

Comment:
Same, the `hypenathed()` thing was strictly for sqlx

Reply:
wasn't aware that this was only for sqlx, fixed